configure.ac

```diff
@@ -194,14 +194,19 @@ LIBS="$saved_LIBS"
 
 # OS detection
 
+AM_CONDITIONAL([KNET_LINUX], [false])
+AM_CONDITIONAL([KNET_BSD], [false])
+
 AC_MSG_CHECKING([for os in ${host_os}])
 case "$host_os" in
 	*linux*)
 		AC_DEFINE_UNQUOTED([KNET_LINUX], [1], [Compiling for Linux platform])
+		AM_CONDITIONAL([KNET_LINUX], [true])
```

Comment:
Hi Fabio, are you sure that redefining Automake conditionals is allowed? The manual says:
>  If AM_CONDITIONAL is run conditionally (e.g., in a shell if statement), then the result will confuse automake.

Reply:
it did run properly on all the tests I did. We can also change to run only once.